### PR TITLE
Remove unsanitized nulls from input strings columns in reduction gtests

### PR DESCRIPTION
**Pull request by davidwendt**
## Description
Input strings column containing unsanitized nulls may result in undefined behavior.
This PR fixes the input data to not include string characters in null rows in gtests for `REDUCTION_TESTS`.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by bdice — APPROVED** [review]

File: cpp/tests/reductions/scan_tests.cpp [hunk 1]
```diff
@@ -625,12 +626,12 @@ TEST_F(StructScanTest, StructScanMinMaxWithNulls)
   auto const input = [] {
     auto child1 = STRINGS_CW{{"año",
                               "bit",
-                              "₹1" /*null*/,
+                              "" /*null*/,
                               "aaa" /*NULL*/,
```
Comment (bdice) [inline cpp/tests/reductions/scan_tests.cpp:630]: Can we make these match case? null/NULL?

**Review by bdice — CHANGES_REQUESTED** [review]
I noticed a small issue after approving. We are mixing null/NULL in the same tests. Can we be consistent?

File: cpp/tests/reductions/scan_tests.cpp (hunk 1 above)
Comment (bdice) [reply cpp/tests/reductions/scan_tests.cpp:630]: Wow, sorry for missing that. I would prefer `/* parent null */` and `/* child null */`, but that's not a blocker. I'll approve this.

**Review by bdice — APPROVED** [review]

File: cpp/tests/reductions/scan_tests.cpp (hunk 1 above)
Comment (davidwendt) [reply cpp/tests/reductions/scan_tests.cpp:630]: I agree. There are not too many of these. I can change this as part of this PR.

File: cpp/tests/reductions/list_rank_test.cpp [hunk 2]
```diff
@@ -131,7 +131,7 @@ TEST_F(ListRankScanTest, ListOfStruct)
      false,
      false}};
   auto col2 = cudf::test::strings_column_wrapper{
-    {"x", "x", "a", "a", "b", "b", "a", "b", "a", "b", "a", "c", "a", "c", "a", "c", "b", "b"},
+    {"x", "x", "a", "a", "b", "", "a", "b", "a", "b", "a", "c", "a", "c", "", "", "b", "b"},
```
Comment (karthikeyann) [inline cpp/tests/reductions/list_rank_test.cpp:134]: nit: I typically have a `constexpr auto null = "";`, and use this `null` in initializer list. (to differentiate between empty string and nulls.

**Review by karthikeyann — APPROVED** [review]

**Pull request merged**